modules/salt/pages/large-scale-tuning.adoc

```diff
@@ -0,0 +1,524 @@
+| Notes                |
+| More information     | `man rhn.conf`
+|===
+
+
+
+==== `org.quartz.threadPool.threadCount`
+
+[cols="1,1"]
+|===
+| Description          | The number of Taskomatic worker threads.
+                         Increasing this value allows Taskomatic to serve more clients in parallel.
+| Tune when            | Client count increases significantly
+| Value default        | 20
+| Value recommendation | 20-200
+| Location             | [path]``/etc/rhn/rhn.conf``
+| After changing       | Check `hibernate.c3p0.max_size` and `thread_pool` for adjustment
+| Notes                |
+| More information     | http://www.quartz-scheduler.org/documentation/2.4.0-SNAPSHOT/configuration.html
+|===
+
+
+
+=== `org.quartz.scheduler.idleWaitTime`
+
+[cols="1,1"]
+|===
+| Description          | Cycle time for Taskomatic.
+                         Decreasing this value lowers the latency of Taskomatic.
+| Tune when            | Client count is in the thousands.
+| Value default        | 5000 ms
+| Value recommendation | 1000-5000 ms
+| Location             | [path]``/etc/rhn/rhn.conf``
+| After changing       |
+| Notes                |
+| More information     | http://www.quartz-scheduler.org/documentation/2.4.0-SNAPSHOT/configuration.html
+|===
+
+
+
+=== `MinionActionExecutor.parallel_threads`
+
+[cols="1,1"]
+|===
+| Description          | Number of Taskomatic threads dedicated to sending commands to Salt clients as a result of Actions being executed.
+| Tune when            | Client count is in the thousands.
+| Value default        | 1
+| Value recommendation | 1-10
+| Location             | [path]``/etc/rhn/rhn.conf``
+| After changing       |
+| Notes                | Full parameter name is `taskomatic.com.redhat.rhn.taskomatic.task.MinionActionExecutor.parallel_threads`
+| More information     |
+|===
+
+
+
+=== `rhn-search.java.maxmemory`
+
+[cols="1,1"]
+|===
+| Description          | The maximum amount of memory that the `rhn-search` service can use.
+| Tune when            | Client count increases significantly, and `OutOfMemoryException` errors appear in `journalctl -u rhn-search`.
+| Value default        | 512 MiB
+| Value recommendation | 512-4096 MiB
+| Location             | [path]``/etc/rhn/rhn.conf``
+| After changing       | Check the Memory Usage section in this document.
+| Notes                |
+| More information     |
+|===
+
+
+
+=== `shared_buffers`
+
+[cols="1,1"]
+|===
+| Description          | The amount of memory reserved for PostgreSQL shared buffers, which contain caches of database tables and index data.
+| Tune when            | RAM changes
+| Value default        | 25% of total RAM
+| Value recommendation | 25-40% of total RAM
+| Location             | [path]``/var/lib/pgsql/data/postgresql.conf``
+| After changing       | Check the Memory Usage section in this document.
+| Notes                | The value is formatted as `<N>MB`, where `<N>` is the number of MiB to allocate.
```

Comment:
```suggestion
| Notes                | The value is formatted as `<N>MiB`, where `<N>` is the number of MiB to allocate.
```
General remark: It looks as if we should switch to MiB and GiB where MB and GB isn't meant.

Reply:
In this case I am talking about the formatting of the configuration option - ie. the exact characters that have to be in `postgresql.conf` - so `MiB` will just not work here.

Reply:
I'm always a bit torn on using MiB and GiB. They're technically more correct, but also less frequently used and therefore a source of confusion (I've literally had people raise bugs because they thought the i was a typo). For that reason, I tend to use MB and GB by default, and only use MiB and GiB where it would be incorrect not to.